core.spec.md

```diff
@@ -0,0 +1,501 @@
+
+If a processor chooses to activate support for a feature, the processor MUST activate an implementation which can [satisfy](#sec-Satisfaction) the version required by the document.
+
+
+# Extensibility
+
+Any directives, scalars, enums, input objects, and other schema elements defined by specs MAY be **extensible**. This means that their definitions within a core schema document do not have to exactly match definitions provided by specs. Instead, definitions should match all usages within the document.
+
+Standard extensibility rules:
+- **Directives** may be defined as taking additional arguments, provided those arguments are [prefixed](#sec-Prefixing) with the name of a document feature followed by two underscores ({"__"}). These arguments are interpreted as metadata-on-metadata, and are routed to the declaring spec for interpretation
+- **Enums** defined in features may include values of any name. For extensible enums, the space of **unprefixed** names belongs to the document. The space of **prefixed** names belongs to the spec with that prefix, and may be used to pass data between specs.
+- **Input Objects** may be defined with additional fields, provided those fields are [prefixed](#sec-Prefixing) with the name of a feature followed by two underscores.
+
+# Processing Schemas
+
+```mermaid diagram
+graph LR
+  schema(["📄  Input Schema"]):::file-->proc("🤖 &nbsp;Processor")
+  proc-->output(["📄  Output Schema"]):::file
+  classDef file fill:none,color:#eee;
+  style proc fill:none,stroke:fuchsia,color:fuchsia;
+```
+
+A common use case is that of a processor which consumes a valid input schema and generates an output schema.
+
+The general guidance for processor behavior is: don't react to what you don't understand.
+
+Specifically, processors:
+  - SHOULD pass through {@core} directives which reference unknown feature URLs
+  - SHOULD pass through prefixed directives, types, and other schema elements  
+  - SHOULD pass through directives which are not [associated with](#AssignFeatures) a {@core} feature
+
+Processors MAY accept configuration which overrides these default behaviors.
+
+Additionally, processors which prepare the schema for final public consumption MAY choose to eliminate all unknown directives and prefixed types in order to hide schema implementation details within the published schema. This will impair the operation of tooling which relies on these directives—such tools will not be able to run on the output schema, so the benefits and costs of this kind of information hiding should be weighed carefully on a case-by-case basis.
+
+# Validations &amp; Algorithms
+
+## Bootstrapping (validates Has Core Feature)
+
+Determine the name of the core specification within the document *or* fail the *Has Core Feature* validation.
+
+It is possible to [rename the core feature](#sec-Renaming-core-itself) within a document. This process determines the actual name for the core feature if one is present.
+
+Bootstrap(document) :
```

Comment:
SUBSTANTIVE: Earlier we say "The first {@core} directive on the schema MUST reference the core spec itself, i.e. this document." but this algorithm does not require the bootstrap schema to be the first one. Which is it?